crates/sui-core/src/authority_server.rs

```diff
@@ -572,8 +549,15 @@ impl ValidatorService {
     ) -> Result<tonic::Response<SubmitCertificateResponse>, tonic::Status> {
         let epoch_store = self.state.load_epoch_store_one_call_per_task();
         let certificate = request.into_inner();
-        // The call to digest() assumes the transaction is valid, so we need to verify it first.
-        certificate.validity_check(epoch_store.protocol_config())?;
+        // Being double cautious to also check this for certificates even though we have
+        // already checked when signing transactions.
+        Self::transaction_validity_check(&epoch_store, certificate.data()).tap_err(|err| {
+            // TODO: This is an invariant violation.
```

Comment:
what's the thing to-do here? (and below)

Reply:
Eventually I want to make it an invariant violation metric.

Reply:
nit/request, Can you make the todo say "TODO: make an invariant violation metric" in that case, just for posterity?